src/components/ChatEntry.js

```diff
@@ -1,22 +1,53 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const message = props.body;
+  const formattedTimeStamp = <TimeStamp time={props.timeStamp}></TimeStamp>;
+  const sender = props.sender;
+  const changeLikeStatus = () => {
+    const updatedChat = {
+      id: props.id,
+      sender: sender,
+      body: message,
+      timeStamp: props.timeStamp,
+      liked: !props.liked,
+    };
+    props.onUpdate(updatedChat);
+  };
```

Comment:
This works just fine, and it's an example we show in Learn. 

But consider that `onUpdate` has been given the responsibility to update the likes, right? Let `onUpdate` take care of it by reassigning `liked`. All `changeLikeStatus` should do is grab the `id` that `onUpdate` needs and pass it through.

Reply:
Be aware that this refactoring would mean you would have to adjust `updateChatMessageData` a bit to accept only an id instead of an entire object.